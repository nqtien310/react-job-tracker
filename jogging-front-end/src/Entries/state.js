import Rx from 'rxjs/Rx'
import api from '../api'
import { combineReducers } from 'redux'
import { push } from 'react-router-redux'
import setErrorMessage from '../actions/setErrorMessage.js'
import {reset} from 'redux-form'
//constants
export const ENTRY_FORM_CREATE = 'ENTRY_FORM_CREATE'
export const ENTRY_FORM_EDIT   = 'ENTRY_FORM_EDIT'
export const ENTRY_FORM_HIDE   = 'ENTRY_FORM_HIDE'
export const ENTRY_DELETE      = 'ENTRY_DELETE'
export const ENTRY_CREATE      = 'ENTRY_CREATE'
export const ENTRY_UPDATE      = 'ENTRY_UPDATE'
export const ENTRIES_FETCH     = 'ENTRIES_FETCH'
export const ENTRIES_FETCHED   = 'ENTRIES_FETCHED'

//actions
export const deleteEntry = (userId, entryId) => {
  return {
    type: ENTRY_DELETE,
    payload:{
      userId: userId,
      id: entryId
    }
  }
}

export const fetchEntries = (userId) => ({type: ENTRIES_FETCH, payload: userId})
export const showCreateForm = () => ({type: ENTRY_FORM_CREATE})
export const showEditForm = (entry) => ({type: ENTRY_FORM_EDIT, payload: entry})
export const hideForm = (entry) => ({type: ENTRY_FORM_HIDE})
export const createEntry = (userId, params) => {
  return {
    type: ENTRY_CREATE,
    payload: {
      userId: userId,
      params: params
    }
  }
}

export const updateEntry = (userId, id, params) => {
  return {
    type: ENTRY_UPDATE,
    payload: {
      userId: userId,
      id: id,
      params: params
    }
  }
}

export const fetchedEntries = (response) => {
  return {
    type: ENTRIES_FETCHED,
    payload: response.data
  }
}

//reducers
function listReducer(state=[], action){
  switch(action.type){
    case ENTRIES_FETCHED: {
      return action.payload
    }default: {
      return state
    }
  }
}

function formReducer(state={}, action){
  switch(action.type){
    case ENTRY_FORM_CREATE: {
      return {action: 'create'}
    }case ENTRY_FORM_EDIT: {
      return {action: 'edit', editing: action.payload}
    }case ENTRY_FORM_HIDE: {
      return {}
    }default: {
      return state
    }
  }
}

export const entryReducer = combineReducers({
  list: listReducer,
  form: formReducer
})

//Epics

export const deleteEntryEpic = (action$) => {
  return action$.ofType(ENTRY_DELETE).switchMap( action => {
    return Rx.Observable.fromPromise(
      api.delete(`users/${action.payload.userId}/entries/${action.payload.id}`)
    ).flatMap(response => [fetchEntries(action.payload.userId), hideForm()])
  })
}

export const updateEntryEpic = (action$) => {
  return action$.ofType(ENTRY_UPDATE).switchMap( action => {
    return Rx.Observable.fromPromise(
      api.put(`users/${action.payload.userId}/entries/${action.payload.id}`, action.payload.params)
    ).flatMap(response => [fetchEntries(action.payload.userId), hideForm()])
  })
}

export const fetchEntriesEpic = (action$) => {
  return action$.ofType(ENTRIES_FETCH).switchMap( action => {
    return Rx.Observable.fromPromise(
      api.get(`users/${action.payload}/entries`)
    ).map(response => fetchedEntries(response))
  })
}

export const createEntryEpic = (action$) => {
  return action$.ofType(ENTRY_CREATE).switchMap( action => {
    return Rx.Observable.fromPromise(
      api.post(`users/${action.payload.userId}/entries`, action.payload.params)
    ).flatMap(() => [fetchEntries(action.payload.userId), hideForm()]
    ).catch(error => Rx.Observable.of(setErrorMessage(error)))
  })
}


//https://github.com/redux-observable/redux-observable/tree/master/examples/navigation/epics
