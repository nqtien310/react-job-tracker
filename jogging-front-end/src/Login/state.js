import { fetchMyUser } from '../epics/fetchMyUserEpic'
import Auth from '../auth'
import 'rxjs/add/operator/switchMap';
import Rx from 'rxjs/Rx'
import api from '../api'
import setErrorMessage from '../actions/setErrorMessage.js'

export const LOGIN  = 'LOGIN'
export const SETTOKEN = 'SETTOKEN'

export const submitLogin = params => ({type: LOGIN, payload: params})
export const setToken    = params => ({type: SETTOKEN, payload: params})


export const loginEpic = (action$) => {
  return action$.ofType(LOGIN).switchMap(action =>{
    return Rx.Observable.fromPromise(
      api.post('login', {user: action.payload})
    ).flatMap(response =>
        [setToken(response), fetchMyUser()]
      ).catch(error => setErrorMessage(error))
  })
}

