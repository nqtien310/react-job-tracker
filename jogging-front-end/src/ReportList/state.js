import Auth from '../auth'
import api from '../api'
import Rx from 'rxjs/Rx'
import setErrorMessage from '../actions/setErrorMessage.js'
import { combineReducers } from 'redux'

export const REPORTS_FETCH = 'REPORTS_FETCH'
export const REPORTS_FETCHED = 'REPORTS_FETCHED'

export const fetchReports = (userId) => {
  return {
    type: REPORTS_FETCH,
    payload:{
      userId: userId
    }
  }
}

export const fetchedReports = (response) => {
  return {
    type: REPORTS_FETCHED,
    payload: response.data
  }
}

const listReducer = (state=[], action) => {
  switch(action.type){
    case REPORTS_FETCHED: {
      return action.payload
    }default: {
      return state
    }
  }
}

export const reportReducer = combineReducers({
  list: listReducer
})

export const fetchReportsEpic = (action$) => {
  return action$.ofType(REPORTS_FETCH).switchMap((action) => {
    return Rx.Observable.fromPromise(
      api.get(`/users/${action.payload.userId}/weekly_summaries`)
    ).flatMap(response => [fetchedReports(response)])
  })
}
