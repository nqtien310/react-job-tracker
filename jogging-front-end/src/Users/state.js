import { combineReducers } from 'redux'
import { push } from 'react-router-redux'
import Rx from 'rxjs/Rx'
import api from '../api'
import setErrorMessage from '../actions/setErrorMessage'

export const USERS_FETCH = 'USERS_FETCH'
export const USER_FETCH = 'USER_FETCH'

export const USER_FETCHED = 'USER_FETCHED'
export const USERS_FETCHED = 'USERS_FETCHED'
export const USER_CREATE  = 'SUBMIT_CREATE_USER'

export const USER_UPDATE = 'USER_UPDATE'

export const fetchUsers = () => ({type: USERS_FETCH})
export const fetchUser = (userId) => ({type: USER_FETCH, payload: userId})
export const fetchedUser = (response) => ({type: USER_FETCHED, payload: response.data})
export const fetchedUsers = (response) => ({type: USERS_FETCHED, payload: response.data})
export const createUser = params => ({type: USER_CREATE, payload: params})
export const updateUser = (userId, params) => ({type: USER_UPDATE, payload: {params: params, userId: userId}})

function listReducer(state=[], action){
  switch(action.type) {
    case USERS_FETCHED: {
      return action.payload
    }default: {
      return state
    }
  }
}

function selectedReducer(state=null, action){
  switch(action.type) {
    case USER_FETCHED: {
      return action.payload
    }default: {
      return state
    }
  }
}

export const userReducer = combineReducers({
  list: listReducer,
  selected: selectedReducer
})

export const fetchUsersEpic = (action$) => {
  return action$.ofType(USERS_FETCH).switchMap((action) => {
    return Rx.Observable.fromPromise(api.get("/users"))
      .flatMap(response => [fetchedUsers(response)])
  })
}

export const fetchUserEpic = (action$) => {
  return action$.ofType(USER_FETCH).switchMap((action) => {
    return Rx.Observable.fromPromise(api.get(`/users/${action.payload}`))
      .flatMap(response => [fetchedUser(response)])
  })
}

export const createUserEpic = (action$) => {
  return action$.ofType(USER_CREATE).switchMap(action =>{
    return Rx.Observable.fromPromise(
      api.post('/register', {user: action.payload})
    ).flatMap(response => [push('/login'), setErrorMessage(null)]
    ).catch(error => Rx.Observable.of(setErrorMessage(error)))
  })
}

export const updateUserEpic = (action$) => {
  return action$.ofType(USER_UPDATE).switchMap(action =>{
    return Rx.Observable.fromPromise(
      api.put(`/users/${action.payload.userId}`, {user: action.payload.params})
    ).flatMap(response => [push('/'), setErrorMessage(null)]
    ).catch(error => Rx.Observable.of(setErrorMessage(error)))
  })
}

