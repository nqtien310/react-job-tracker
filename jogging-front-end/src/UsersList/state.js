import { combineReducers } from 'redux'
import Rx from 'rxjs/Rx'
import api from '../api'

export const USERS_FETCH = 'USERS_FETCH'
export const USERS_FETCHED = 'USERS_FETCHED'

export const fetchUsers = () => ({type: USERS_FETCH})
export const fetchedUsers = (response) => ({type: USERS_FETCHED, payload: response.data})

function listReducer(state=[], action){
  switch(action.type) {
    case USERS_FETCHED: {
      return action.payload
    }default: {
      return state
    }
  }
}

export const userReducer = combineReducers({
  list: listReducer
})

export const fetchUsersEpic = (action$) => {
  return action$.ofType(USERS_FETCH).switchMap((action) => {
    return Rx.Observable.fromPromise(api.get("/users"))
      .flatMap(response => [fetchedUsers(response)])
  })
}
