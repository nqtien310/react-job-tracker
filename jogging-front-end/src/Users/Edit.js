import React from 'react';
import Template from '../Template'
import Form from './Form'
import { fetchUser, updateUser } from './state'
import { connect } from 'react-redux'

class EditForm extends React.Component {
  constructor(props){
    super(props)
    this.userId = this.props.match.params.userId
  }

  componentWillMount() {
    this.props.fetchUser(this.userId)
  }

  onSubmit = (params) => {
    this.props.updateUser(this.userId, params)
  }

  render() {
    return (
      <Form onSubmit={this.onSubmit} initialValues={this.props.selectedUser}/>
    )
  }
}

function mapStateToProps(state){
  return {
    selectedUser: state.user.selected
  }
}

export default connect(mapStateToProps, {
  fetchUser,
  updateUser
})(EditForm)
