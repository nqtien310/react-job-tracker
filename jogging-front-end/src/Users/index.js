import React from 'react';
import { connect } from 'react-redux'
import { fetchUsers, deleteUser } from './state'
import { Link } from 'react-router-dom'

class UsersList extends React.Component{
  componentWillMount(){
    this.props.fetchUsers()
  }

  onDelete(userId) {
    this.props.deleteUser(userId)
  }

  renderEntriesActions(user) {
    if(this.props.myUser.role == "admin"){
      return (
        <td className="actions">
          <a onClick={()=> this.onDelete(user.id)} className="btn btn-danger">
            <i className="fa fa-trash"/>
          </a>
          <Link to={`/users/${user.id}/edit`} className="btn btn-warning">
            <i className="fa fa-edit"/>
          </Link>
          <Link to={`/users/${user.id}/entries`} className="btn btn-primary">
            <i className="fa fa-calendar"/>
          </Link>
          <Link to={`/users/${user.id}/report`} className="btn btn-primary">
            <i className="fa fa-bar-chart"/>
          </Link>
        </td>
      )
    }else {
      return (<td className="actions">
        <a onClick={()=> this.onDelete(user.id)} className="btn btn-danger">
          <i className="fa fa-trash"/>
        </a>
        <Link to={`/users/${user.id}/edit`} className="btn btn-warning">
          <i className="fa fa-edit"/>
        </Link>
      </td> )
    }
  }

  renderUsers() {
    return this.props.users.map(user => {
      return(
        <tr key={user.id}>
          <td> {user.email} </td>
          <td> {user.full_name} </td>
          {this.renderEntriesActions(user)}
        </tr>
      )
    })
  }

  render() {
    if(this.props.users.length == 0){
      return (
        <div> <div> No users created </div>

          <div className="action-footers">
            <Link to="/users/new" className="btn btn-primary">
              <i className="fa fa-plus"/>
            </Link>
          </div>
        </div>
      )
    }else {
      return (
        <div>
          <table className="table">
            <thead>
              <tr>
                <th>Email</th>
                <th>Full Name</th>
                <th>Action</th>
              </tr>
            </thead>

            <tbody>
              {this.renderUsers()}
            </tbody>
          </table>

          <div className="action-footers">
            <Link to="/users/new" className="btn btn-primary">
              <i className="fa fa-plus"/>
            </Link>
          </div>
        </div>
      )
    }
  }
}

function mapStateToProps(state){
  return {
    users: state.user.list
  }
}
export default connect(mapStateToProps, {
  fetchUsers,
  deleteUser
})(UsersList)
