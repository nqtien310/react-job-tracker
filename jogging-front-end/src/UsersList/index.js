import React from 'react';
import { connect } from 'react-redux'
import { fetchUsers } from './state'

class UsersList extends React.Component{
  componentWillMount(){
    this.props.fetchUsers()
  }

  onDelete() {
    console.log("DELETE")
  }

  onEdit() {
    console.log("EDIT")
  }

  renderUsers() {
    return this.props.users.map(user => {
      return(
        <tr key={user.id}>
          <td> {user.email} </td>
          <td> {user.full_name} </td>
          <td className="actions">
            <a onClick={()=> this.onDelete(user.id)} className="btn btn-danger">Delete</a>
            <a onClick={()=> this.onEdit(user.id)} className="btn btn-warning">Edit</a>
          </td>
        </tr>
      )
    })
  }

  render() {
    return (
      <table className="table">
        <thead>
          <tr>
            <th>Email</th>
            <th>Full Name</th>
            <th>Action</th>
          </tr>
        </thead>

        <tbody>
          {this.renderUsers()}
        </tbody>
      </table>
    )
  }
}

function mapStateToProps(state){
  return {
    users: state.user.list
  }
}
export default connect(mapStateToProps, {
  fetchUsers
})(UsersList)
