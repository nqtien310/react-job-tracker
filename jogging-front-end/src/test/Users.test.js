import React from 'react';
import renderer from 'react-test-renderer'
import LoginForm from '../Login'
import RenderComponent from './RenderComponent'
import { store } from './RenderComponent'
import UsersList from '../Users'
import { fetchUsers, createUser, updateUser, deleteUser } from '../Users/state'
import { stubApiSuccess } from './ApiStubs.js'

it('No User created', (done) => {
  stubApiSuccess("get", [])

  const component = renderer.create(
    <RenderComponent>
      <UsersList myUser={{id: 1}}/>
    </RenderComponent>
  )

  setTimeout( () => {
    expect(component).toMatchSnapshot()
    done()
  }, 50)
});

it('With Users', (done) => {
  let data = [{"id":43,"email":"nqtien310@gmail.com","full_name":"Tien Nguyen","role":"user"}]
  stubApiSuccess("get", data)

  const component = renderer.create(
    <RenderComponent>
      <UsersList myUser={{id: 1}}/>
    </RenderComponent>
  )

  setTimeout( () => {
    expect(component).toMatchSnapshot()
    done()
  }, 50)
});

it('Delete', (done) => {
  let data = [{"id":43,"email":"nqtien310@gmail.com","full_name":"Tien Nguyen","role":"user"}]
  stubApiSuccess("get", data)

  const component = renderer.create(
    <RenderComponent>
      <UsersList myUser={{id: 1}}/>
    </RenderComponent>
  )

  setTimeout( () => {
    expect(component).toMatchSnapshot()
    let dataAfter = []
    stubApiSuccess("get", dataAfter)
    stubApiSuccess("delete", null)
    store.dispatch(deleteUser(43))

    setTimeout ( () => {
      expect(component).toMatchSnapshot()
      done()
    }, 50)
  }, 50)
});

it('Update', (done) => {
  let data = [{"id":43,"email":"nqtien310@gmail.com","full_name":"Tien Nguyen","role":"user"}]
  stubApiSuccess("get", data)

  const component = renderer.create(
    <RenderComponent>
      <UsersList myUser={{id: 1}}/>
    </RenderComponent>
  )

  setTimeout( () => {
    expect(component).toMatchSnapshot()
    let dataAfter = [{"id":43,"email":"updated@gmail.com","full_name":"Tien Nguyen","role":"user"}]
    stubApiSuccess("get", dataAfter)
    stubApiSuccess("put", {})
    store.dispatch(updateUser(43,{}))
    store.dispatch(fetchUsers())

    setTimeout ( () => {
      expect(component).toMatchSnapshot()
      done()
    }, 50)
  }, 50)
});

it('Create', (done) => {
  let data = [{"id":43,"email":"nqtien310@gmail.com","full_name":"Tien Nguyen","role":"user"}]
  stubApiSuccess("get", data)

  const component = renderer.create(
    <RenderComponent>
      <UsersList myUser={{id: 1}}/>
    </RenderComponent>
  )

  setTimeout( () => {
    expect(component).toMatchSnapshot()
    let dataAfter = [
      {"id":43,"email":"nqtien31@gmail.com","full_name":"Tien Nguyen","role":"user"},
      {"id":44,"email":"new@gmail.com","full_name":"New","role":"user"}
    ]
    stubApiSuccess("get", dataAfter)
    stubApiSuccess("post", {})
    store.dispatch(createUser(43,{}))
    store.dispatch(fetchUsers())

    setTimeout ( () => {
      expect(component).toMatchSnapshot()
      done()
    }, 50)
  }, 50)
});
