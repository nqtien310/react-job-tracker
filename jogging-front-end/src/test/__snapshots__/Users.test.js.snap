// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create 1`] = `
<div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Email
        </th>
        <th>
          Full Name
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
           
          nqtien310@gmail.com
           
        </td>
        <td>
           
          Tien Nguyen
           
        </td>
        <td
          className="actions"
        >
          <a
            className="btn btn-danger"
            onClick={[Function]}
          >
            <i
              className="fa fa-trash"
            />
          </a>
          <a
            className="btn btn-warning"
            href="/users/43/edit"
            onClick={[Function]}
          >
            <i
              className="fa fa-edit"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="action-footers"
  >
    <a
      className="btn btn-primary"
      href="/users/new"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </a>
  </div>
</div>
`;

exports[`Create 2`] = `
<div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Email
        </th>
        <th>
          Full Name
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
           
          nqtien31@gmail.com
           
        </td>
        <td>
           
          Tien Nguyen
           
        </td>
        <td
          className="actions"
        >
          <a
            className="btn btn-danger"
            onClick={[Function]}
          >
            <i
              className="fa fa-trash"
            />
          </a>
          <a
            className="btn btn-warning"
            href="/users/43/edit"
            onClick={[Function]}
          >
            <i
              className="fa fa-edit"
            />
          </a>
        </td>
      </tr>
      <tr>
        <td>
           
          new@gmail.com
           
        </td>
        <td>
           
          New
           
        </td>
        <td
          className="actions"
        >
          <a
            className="btn btn-danger"
            onClick={[Function]}
          >
            <i
              className="fa fa-trash"
            />
          </a>
          <a
            className="btn btn-warning"
            href="/users/44/edit"
            onClick={[Function]}
          >
            <i
              className="fa fa-edit"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="action-footers"
  >
    <a
      className="btn btn-primary"
      href="/users/new"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </a>
  </div>
</div>
`;

exports[`Delete 1`] = `
<div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Email
        </th>
        <th>
          Full Name
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
           
          nqtien310@gmail.com
           
        </td>
        <td>
           
          Tien Nguyen
           
        </td>
        <td
          className="actions"
        >
          <a
            className="btn btn-danger"
            onClick={[Function]}
          >
            <i
              className="fa fa-trash"
            />
          </a>
          <a
            className="btn btn-warning"
            href="/users/43/edit"
            onClick={[Function]}
          >
            <i
              className="fa fa-edit"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="action-footers"
  >
    <a
      className="btn btn-primary"
      href="/users/new"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </a>
  </div>
</div>
`;

exports[`Delete 2`] = `
<div>
   
  <div>
     No users created 
  </div>
  <div
    className="action-footers"
  >
    <a
      className="btn btn-primary"
      href="/users/new"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </a>
  </div>
</div>
`;

exports[`No User created 1`] = `
<div>
   
  <div>
     No users created 
  </div>
  <div
    className="action-footers"
  >
    <a
      className="btn btn-primary"
      href="/users/new"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </a>
  </div>
</div>
`;

exports[`Update 1`] = `
<div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Email
        </th>
        <th>
          Full Name
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
           
          nqtien310@gmail.com
           
        </td>
        <td>
           
          Tien Nguyen
           
        </td>
        <td
          className="actions"
        >
          <a
            className="btn btn-danger"
            onClick={[Function]}
          >
            <i
              className="fa fa-trash"
            />
          </a>
          <a
            className="btn btn-warning"
            href="/users/43/edit"
            onClick={[Function]}
          >
            <i
              className="fa fa-edit"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="action-footers"
  >
    <a
      className="btn btn-primary"
      href="/users/new"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </a>
  </div>
</div>
`;

exports[`Update 2`] = `
<div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Email
        </th>
        <th>
          Full Name
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
           
          updated@gmail.com
           
        </td>
        <td>
           
          Tien Nguyen
           
        </td>
        <td
          className="actions"
        >
          <a
            className="btn btn-danger"
            onClick={[Function]}
          >
            <i
              className="fa fa-trash"
            />
          </a>
          <a
            className="btn btn-warning"
            href="/users/43/edit"
            onClick={[Function]}
          >
            <i
              className="fa fa-edit"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="action-footers"
  >
    <a
      className="btn btn-primary"
      href="/users/new"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </a>
  </div>
</div>
`;

exports[`With Users 1`] = `
<div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Email
        </th>
        <th>
          Full Name
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
           
          nqtien310@gmail.com
           
        </td>
        <td>
           
          Tien Nguyen
           
        </td>
        <td
          className="actions"
        >
          <a
            className="btn btn-danger"
            onClick={[Function]}
          >
            <i
              className="fa fa-trash"
            />
          </a>
          <a
            className="btn btn-warning"
            href="/users/43/edit"
            onClick={[Function]}
          >
            <i
              className="fa fa-edit"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="action-footers"
  >
    <a
      className="btn btn-primary"
      href="/users/new"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </a>
  </div>
</div>
`;
